src/main/java/in/twizmwaz/cardinal/module/modules/kit/KitCollection.java

```diff
@@ -0,0 +1,101 @@
+package in.twizmwaz.cardinal.module.modules.kit;
+
+import in.twizmwaz.cardinal.GameHandler;
+import in.twizmwaz.cardinal.event.CycleCompleteEvent;
+import in.twizmwaz.cardinal.module.modules.filter.FilterModule;
+import in.twizmwaz.cardinal.module.modules.filter.FilterState;
+import org.bukkit.Material;
+import org.bukkit.entity.EnderPearl;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class KitCollection implements KitRemovable {
```

Comment:
KitComplete?

Also, perhaps consider moving the boolean values to their own kit objects, it makes the most sense to fully modularize it. It may also be worth it to store actual references to the parent kits rather than just a string.